cache/http_test.go

```diff
@@ -150,3 +151,59 @@ func TestUploadSameFileConcurrently(t *testing.T) {
 
 	wg.Wait()
 }
+
+func TestArtifactInfoFromUrl(t *testing.T) {
+	{
+		info, err := artifactInfoFromUrl("invalid/url", "")
+		if info != nil || err == nil {
+			t.Error("Failed to reject an invalid URL")
+		}
+	}
+
+	{
+		info, err := artifactInfoFromUrl(
+			"cas/fec3be77b8aa0d307ed840581ded3d114c86f36d4914c81e33a72877020c0603",
+			"BASEDIR")
+		if err != nil {
+			t.Error("Failed to parse a valid CAS URL")
+		}
+		if !info.verifyHash {
+			t.Error("CAS requests should have verifyHash == true")
+		}
+		if !strings.HasPrefix(info.hash, "fec3") || !strings.HasSuffix(info.hash, "0603") {
+			t.Error("Hash parsed incorrectly")
+		}
+		if !strings.HasPrefix(info.filePath, "BASEDIR/cas/fec3") ||
+			!strings.HasSuffix(info.filePath, "0603") {
+			t.Error("File path constructed incorrectly")
```

Comment:
Log expected and actual? It’s always nice to know from an error message what’s wrong.

Reply:
How do you feel about one of those assert-like packages for go?